Always position URL bar caret at the beginning when focusing

keianhzo opened the pull request:
Fixes #476 Always position URL bar caret at the beginning when focusing

cvan commented:
P.S. I spent a few minutes to see if I could easily add placeholder text when the URL bar is empty and not focused and on the Home page/`about:` pages. Issue #358. Here's my WIP branch: https://github.com/MozillaReality/FirefoxReality/compare/search-placeholder?expand=1

It's close, but I'm getting a bit confused about how to store the real URL vs. the placeholder, and whether this logic should live in `NavigationBarWidget` instead of in `NavigationURLBar`. If there are more pressing items, we can wait. When you launch FxR today, the URL bar is empty and it looks awkward and confusing, since there's no search icon or anything else in the URL bar. Let me know if either of you have ideas for a quick fix.

keianhzo commented:
@cvan I've added a fix for that in #522 I've implemented the placeholder text using a hint that automatically handles the placeholder text when the text is empty.
